C++/CMakeLists.txt

```diff
@@ -157,6 +157,11 @@ add_executable(shunting_yard
         test/algorithm/string/shunting_yard.cpp)
 target_link_libraries(shunting_yard test_runner)
 
+# Heap's Algorithm
```

Comment:
`# Heap's` (no need to identify it as an algorithm since that's implied by the category)